config/routes.rb

```diff
@@ -1,3 +1,15 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  devise_scope :user do
+    authenticated :user do
+      root to: 'users#show', as: :authenticated_root
+    end
+  
+    unauthenticated :user do
+      root to: 'static_pages#home', as: :unauthenticated_root
```

Comment:
- [ ] **[OPTIONAL]** Nice work have separate routes for authenticated and unauathenticated users, but to get rid of the redundant message 'You need to log in before continuiing', it's better to set the root for unauthenticated to be sign_in page.
```ruby
unauthenticated :user do
    root to: 'devise/sessions#new'
end
```